dds/DCPS/ThreadPool.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DCPS/DdsDcps_pch.h" //Only the _pch include should start with DCPS/
+
+#include "ThreadPool.h"
+
+#include <ace/Guard_T.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+ThreadPool::ThreadPool(size_t count, FunPtr fun, void* arg)
+ : barrier_(count + 1)
+ , fun_(fun)
+ , arg_(arg)
+ , ids_(count, 0)
+{
+  {
+    ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
+    for (size_t i = 0; i < count; ++i) {
+      ACE_Thread::spawn(run, this, THR_NEW_LWP | THR_JOINABLE, 0, &(ids_[i]));
```

Comment:
Is this the same as `spawn_n`?

Reply:
I think I had it that way at first but changed it for some reason, perhaps because the flags weren't defaulted or because I wasn't getting the handles the way I expected.